### DISCUSSION TITLE
[Chapter 06] 로그 설정에 관하여

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#59] [Chapter 06. Building a Web Application Server] [answered]
6.7 장을 통해 스프링 부트 애플리케이션에 로그를 남기는 방법이 5장 정도에 걸쳐 소개되고 있습니다.

그렇다면 실제로 운영되고 있는 애플리케이션이 있고, 하루 트래픽이 100만 건 정도 된다고 가정해 보겠습니다.
그럼 수집되는 로그의 양도 상당할 텐데, 만약 특정 사용자가 오류가 발생했다면 어떻게 해당 사용자의 오류 로그를 탐색해야 할까요?
그리고, 어떻게 로깅을 구성하면 많은 트래픽이 발생하는 애플리케이션이라도 로깅을 빠르고 효율적으로 할 수 있을까요?

**Comment by ahngilwoong** [comment, marked as answer]
이 당시에는 몰랐지만
제가 회사의 운영/개발 업무를 맡으며 로그는 정말 운영의 입장에서는 다다익선이고 제 생각보다 너무나 소중한 존재였습니다.

저때는 몰랐지만 지금 생각해보니 제가 사용하는 시스템과 유사한 상황인 것 같습니다.
100만건 까지는 아니지만 공장 데이터들이 분당 수도없이 이벤트들을 발생시키고 있는 상황에서 오류가 나면 언제 어디서 오류가 나는것인지
빠르게 캐치하고 조치하는게 저의 숙명입니다.

1. 로그 데이터를 효율적으로 관리하는 도구를 사용하면 좋을 것 같습니다 (저희 회사는 Splunk라는 로그 분석 도구를 사용합니다.)
2. 로그의 패턴화가 중요할 것 같습니다. 특정 키워드로 필요한 내용을 빠르게 찾아낼 수 있어야 할 것 같습니다.
3. Splunk에서 따로 세션id, 요청id를 관리하고 있어 특정한 곳에서 오는 흐름들의 로그를 집합해서 볼 수 있습니다.

저는 크게 이 3가지로 정말 로그의 큰 도움을 받고있습니다.
